tests/gar.test.mjs

```diff
@@ -775,329 +510,533 @@ describe('GatewayRegistry', async () => {
           observedEpochCount: 0,
         },
       });
+
+      if (delegateAddresses && expectedDelegates) {
+        var nextMemory = joinNetworkMemory;
+        for (const delegateAddress of delegateAddresses) {
+          const { memory: delegatedMemory } = await delegateStake({
+            memory: joinNetworkMemory,
+            timestamp: STUB_TIMESTAMP,
+            delegatorAddress: delegateAddress,
+            quantity: 500_000_000,
+            gatewayAddress: otherGatewayAddress,
+          }).catch(() => {});
+          nextMemory = delegatedMemory;
+        }
+        const updatedGateway = await getGateway({
+          address: otherGatewayAddress,
+          memory: nextMemory,
+        });
+        assert.deepEqual(
+          Object.keys(updatedGateway.delegates),
+          expectedDelegates,
+        );
+      }
+    }
+
+    it('should allow joining of the network with an allow list', async () => {
+      await allowlistJoinTest({
+        tags: [
+          { name: 'Allow-Delegated-Staking', value: 'allowlist' },
```

Comment:
just a reminder for ourselves - we'll want to update the SDK to support this value (and the new lookup table)